src/neuroconv/tools/nwb_helpers/_models/_base_dataset_models.py

```diff
@@ -0,0 +1,115 @@
+            "For optimized writing speeds and minimal RAM usage, a total size of around 1 GB is recommended."
+        )
+    )
+    # TODO: When using Pydantic v2, wrap h5py._hl.filters.FilterRefBase and numcodecs.abc.Codec with InstanceOf
+    compression_method: Union[str, h5py._hl.filters.FilterRefBase, numcodecs.abc.Codec, None] = Field(
+        default="gzip",
+        description="The specified compression method to apply to this dataset. Set to `None` to disable compression.",
+    )
+    compression_options: Union[Dict[str, Any], None] = Field(
+        default=None, description="The optional parameters to use for the specified compression method."
+    )
+
+    def __str__(self) -> str:
+        """Not overriding __repr__ as this is intended to render only when wrapped in print()."""
+        string = (
+            f"\n{self.dataset_info.location}"
+            f"\n{'-' * len(self.dataset_info.location)}"
+            f"\n  maxshape: {self.dataset_info.full_shape}"
+            f"\n  dtype: {self.dataset_info.dtype}"
+            "\n"
+            f"\n  chunk_shape: {self.chunk_shape}"
+            f"\n  buffer_shape: {self.buffer_shape}"
+            f"\n  compression_method: {self.compression_method}"
+        )
+        if self.compression_options is not None:
+            string += f"\n  compression_options: {self.compression_options}"
+
+        return string
+
+    @root_validator
+    def validate_all_shapes(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        chunk_shape = values["chunk_shape"]
+        buffer_shape = values["buffer_shape"]
+        full_shape = values["dataset_info"].full_shape
+
+        if len(chunk_shape) != len(buffer_shape):
+            raise ValueError(f"{len(chunk_shape)=} does not match {len(buffer_shape)=}!")
+        if len(buffer_shape) != len(full_shape):
+            raise ValueError(f"{len(buffer_shape)=} does not match {len(full_shape)=}!")
+
+        if any(chunk_axis <= 0 for chunk_axis in chunk_shape):
+            raise ValueError(f"Some dimensions of the {chunk_shape=} are less than or equal to zero!")
+        if any(buffer_axis <= 0 for buffer_axis in buffer_shape):
+            raise ValueError(f"Some dimensions of the {buffer_shape=} are less than or equal to zero!")
+
+        if any(chunk_axis > buffer_axis for chunk_axis, buffer_axis in zip(chunk_shape, buffer_shape)):
+            raise ValueError(f"Some dimensions of the {chunk_shape=} exceed the {buffer_shape=})!")
+        if any(buffer_axis > full_axis for buffer_axis, full_axis in zip(buffer_shape, full_shape)):
+            raise ValueError(f"Some dimensions of the {buffer_shape=} exceed the {full_shape=}!")
+
+        if any(buffer_axis % chunk_axis != 0 for chunk_axis, buffer_axis in zip(chunk_shape, buffer_shape)):
+            raise ValueError(f"Some dimensions of the {chunk_shape=} do not evenly divide the {buffer_shape=})!")
+
+        return values
+
+    def get_data_io_keyword_arguments(self):
```

Comment:
Or maybe is not even needed.

Reply:
Oh, I see that in another PR you want to have a uniform method to call this. I need to take a closer look.

Reply:
I suppose it could be; it is indeed required to be used in the `configure_backend` method: https://github.com/catalystneuro/neuroconv/pull/571/files#diff-91679913ea0f31861b6d5e7275fa4a7e03b256d6cd91aa8d82ca505c604d9112R226

Went ahead and made it abstract (that is, required to be defined for all child classes)